plugins/block-sharable-procedures/src/events_procedure_parameter_rename.ts

```
@@ -0,0 +1,131 @@
+   * @returns JSON representation.
+   */
+  toJson(): ProcedureParameterRenameJson {
+    const json = super.toJson() as ProcedureParameterRenameJson;
+    json['oldName'] = this.oldName;
+    return json;
+  }
+
+  /**
+   * Deserializes the JSON event.
+   * @param json The JSON representation of a procedure parameter rename event.
+   * @param workspace The workspace to deserialize the event into.
+   * @returns The new procedure parameter rename event.
+   * @internal
+   */
+  static fromJson(
+      json: ProcedureParameterRenameJson,
+      workspace: Blockly.Workspace
+  ): ProcedureParameterRename {
+    const model = workspace.getProcedureMap().get(json['procedureId']);
+    if (!model) {
+      throw new Error(
+          'Cannot deserialize procedure delete event because the ' +
+          'target procedure does not exist');
+    }
+    const param = findMatchingParameter(
+        workspace, json['procedureId'], json['parameterId']);
+    if (!param) {
+      throw new Error(
+          'Cannot deserialize parameter rename event because the ' +
+          'target parameter does not exist');
+    }
+    return new ProcedureParameterRename(
+        workspace, model, param, json['oldName']);
+  }
+}
+
+/**
+ * Finds the parameter with the given ID in the procedure model with the given
+ * ID, if both things exist.
+ * @param workspace The workspace to search for the parameter.
+ * @param modelId The ID of the model to search for the parameter.
+ * @param paramId The ID of the parameter to search for.
+ * @returns The parameter model that was found.
+ */
+function findMatchingParameter(
+    workspace: Blockly.Workspace,
+    modelId: string,
+    paramId: string
+): Blockly.procedures.IParameterModel|undefined {
+  const procedureModel = workspace.getProcedureMap().get(modelId);
+  if (!procedureModel) {
+    throw new Error(
+        'Cannot rename the parameter of a procedure that does not exist ' +
+        'in the procedure map');
+  }
```

Should this throw or just return undefined? The callers appear to be handling a failure to find the parameter in the model by throwing as well, so perhaps this should just also throw in that case as well and return a non-nullable IParameterModel?

---

Good point! Changed `findMatchingParameter` to throw in the case of no parameter for consistency.